lib/common/schemas.c

```diff
@@ -1066,12 +1069,20 @@ apply_upgrade(const xmlNode *original_xml, int schema_index, gboolean to_logs)
     }
 
     if ((final != NULL) && transform_onleave) {
+        char *dash = NULL;
+
         upgrade = final;
         /* following condition ensured in add_schema_by_version */
         CRM_ASSERT(schema->transform_enter != NULL);
+
         transform_leave = strdup(schema->transform_enter);
+        CRM_ASSERT(transform_leave != NULL);
+
+        dash = strrchr(transform_leave, '-');
+        CRM_ASSERT(dash != NULL); // Make static analysis happy
+
         /* enter -> leave */
-        memcpy(strrchr(transform_leave, '-') + 1, "leave", sizeof("leave") - 1);
+        memcpy(dash + 1, "leave", sizeof("leave") - 1);
```

Comment:
GCC on Fedora 40 is complaining about this for some reason. I get that we're relying on an assumption of what comes after the hyphen, but nothing has meaningfully changed AFAICT. Ah well, I guess we'll have to be even more explicit.
```
22:14:35  In file included from /usr/include/string.h:548,
22:14:35                   from /usr/include/glib-2.0/glib/galloca.h:35,
22:14:35                   from /usr/include/glib-2.0/glib.h:32,
22:14:35                   from ../../include/crm_internal.h:27,
22:14:35                   from schemas.c:10:
22:14:35  In function 'memcpy',
22:14:35      inlined from 'apply_upgrade' at schemas.c:1085:9:
22:14:35  /usr/include/bits/string_fortified.h:29:10: error: 'memcpy' offset [0, 4] is out of the bounds [0, 0] [-Werror=array-bounds=]
22:14:35     29 |   return __builtin___memcpy_chk (__dest, __src, __len,
22:14:35        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
22:14:35     30 |                                  __glibc_objsize0 (__dest));
22:14:35        |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Bah, not worth the hassle, just dropped that commit